files/scripts/sonic-dpu-mgmt-traffic.sh

```diff
@@ -39,20 +49,135 @@ control_forwarding(){
     if [ "$op" = "enable" ]; then
         value=1
     fi
-    echo $value > /proc/sys/net/ipv4/ip_forward
-    echo $value > /proc/sys/net/ipv4/conf/eth0/forwarding
+    if [ "$fw_change" = "enable" ]; then
+        echo $value > /proc/sys/net/ipv4/ip_forward
+        echo $value > /proc/sys/net/ipv4/conf/eth0/forwarding
+    fi
+}
+
+validate_dpus(){
+    local provided_list=("$@")
+    for item1 in "${provided_list[@]}"; do
+        local found=0
+        for item2 in "${dpu_l[@]}"; do
+            if [[ "$item1" = "$item2" ]]; then
+                found=1
+                break
+            fi
+        done
+        if [[ $found -eq 0 ]]; then
+            echo "$item1 is not detected! Please provide proper dpu names list!"
+            exit 1
+        fi
+    done
 }
 
-ctrl_dpu_forwarding(){
+general_validation(){
+    if [ -z "$direction" ]; then
+        echo "Please provide the direction argument (inbound or outboud)"
+        usage
+        exit 1
+    fi
+
+    if [ -z "$operation" ]; then
+        echo "Please provide the operation option (-e or -d)"
+        usage
+        exit 1
+    fi
+}
+
+
+inbound_validation(){
+    #DPU Validation
+    while IFS= read -r line; do
+        dpu_name=$( echo "$line" | sed 's/.*|//;s/"//g')
+        dpu_l+=("$dpu_name")
+    done < <(redis-cli -n 4 keys DPUS*)
+    len1=${#dpu_name[@]}
+    if [ "$len1" -eq 0 ]; then
+        echo "No dpus detected on device!"
+        exit 1
+    fi
+    sorted_dpu_l=($(for item in "${dpu_l[@]}"; do
+        echo "$item"
+    done | sort))
+    if [ -z "$arg_dpu_names" ]; then
+            echo "No DPUs provided!"
+            usage
+            exit 1
+        else
+            if [ "$arg_dpu_names" = "all" ]; then
+                sel_dpu_names=("${sorted_dpu_l[@]}")
+                echo "${#sorted_dpu_l[@]} DPUs detected:"
+                echo "${sorted_dpu_l[@]}"
+            else
+                IFS=',' read -ra sel_dpu_names <<< "$arg_dpu_names"
+                validate_dpus ${sel_dpu_names[@]}
+            fi
+    fi
+    #Port validation
+    IFS=',' read -ra provided_ports <<< "$arg_port_list"
```

Comment:
Can we also have range of valid ports to be used for the DPU management access?

Reply:
There are no "valid" ports as such, The user has to make sure to use ports which are currently not in use by other applications/ host.

Reply:
Could you please check if add_rem_valid_iptable() returns error if already used port is used in this.

Reply:
The iptables rules addition does not seem to fail on adding rules with currently "in use" ports, will add additional validation for this

Reply:
Fixed

Reply:
Thanks for adding the port validation check